### PR TITLE
credit backup implemented. message backup implemented halfway. not very robust

### PR DESCRIPTION
**Pull request by AChFr**

### CODE REVIEW

File: src/clients/checkCredit.js [hunk 1]
```diff
@@ -1,16 +1,29 @@
...
+        console.log("Error while checking balance on main db");
+        try {
+            const existingCreditOnSeconday = await secondaryDatabase.findOne()
+            if ((existingCreditOnSeconday) && (existingCreditOnSeconday.amount - messageCost >= 0)) {
+                return true
+            } else {
+                return false
+            }
+        } catch (err) {
+            throw new Error(("something went wrong. unable to check balance on either database. message aborted"))
```
Comment (Teo-Lopez) [inline src/clients/checkCredit.js:26]: This can be refactored, is too nested and hard to read for an easy operation.

**Comment by AChFr** [comment]
I have actually done that. Don't know why it does not show on the pull
request

File: src/clients/saveMessage.js [hunk 2]
```diff
@@ -1,18 +1,26 @@
...
+
+
   try {
 
-    const doc = await message.save();
+    const doc = await mainDatabase.create(messageParams)
 
-    console.log("Message saved succesfully:", doc);
+    // tried to simplify it the most, but  {..doc} did not work
+    await secondaryDatabase.create({ _id: doc._id, destination: doc.destination, body: doc.body, status: doc.status })
```
Comment (Teo-Lopez) [inline src/clients/saveMessage.js:17]: The documents that come from mongoose bring a lot more info than it seems. The actual data is hidden under the hood so when you destructure a document you don't find what you expect

File: src/clients/topUpCredit.js [hunk 3]
```diff
@@ -1,36 +1,41 @@
...
 
 export default async (creditAmount) => {
   const end = await sync();
-
   try {
-    const existingCredit = await Credit.findOne()
 
-    if (existingCredit) {
+    const existingCreditOnMain = await mainDatabase.findOneAndUpdate({}, { $inc: { amount: creditAmount.amount } },
+      { new: true, upsert: true })
```
Comment (Teo-Lopez) [inline src/clients/topUpCredit.js:13]: I see that you looked into it :)

**Review by Teo-Lopez — COMMENTED** [review]
Good work!

**Pull request merged**